zk/syncer/l1_syncer.go

```diff
@@ -280,7 +283,19 @@ func tryToLogL1QueryBlocks(logPrefix string, current, total, threadNum int, dura
 
 func (s *L1Syncer) getLatestL1Block() (uint64, error) {
 	em := s.getNextEtherman()
-	latestBlock, err := em.BlockByNumber(context.Background(), big.NewInt(rpc.FinalizedBlockNumber.Int64()))
+
+	var blockNumber *big.Int
+
+	switch s.highestBlockType {
+	case "finalized":
+		blockNumber = big.NewInt(rpc.FinalizedBlockNumber.Int64())
+	case "safe":
+		blockNumber = big.NewInt(rpc.SafeBlockNumber.Int64())
+	case "latest":
+		blockNumber = nil
```

Comment:
To understand this line: https://github.com/0xPolygonHermez/cdk-erigon/blob/8d331f1cc37caf756082d0cea543ca49e163d244/ethclient/ethclient.go#L745